### DISCUSSION TITLE
Research paper about entity linking feature

### DISCUSSION THREAD

**Discussion by Joozty** [explosion/spaCy#6565] [Help: Other Questions] [answered]
Hi is there any research paper/in-depth explanation of entity linking functionality. It's a really nice feature and I would really like to understand how it works under the hood. 

I found a [video](https://youtu.be/PW3RJM8tDGo) of Sofie Van Landeghem giving a presentation but I would like to know even more about it.

Thanks for help :)

**Comment by svlandeg** [comment, marked as answer]
Hi, happy to hear the Entity Linking work is useful to you!

There's no paper, and I think the underlying methodology is actually quite straightforward right now.

I'd be happy to give you some additional pointers that may help you with practical applications though.
- We created a video on how to use the EL algorithm in spaCy v2 (on a toy dataset): https://www.youtube.com/watch?v=8u57WSXVpmw. The corresponding code can be found here: https://github.com/explosion/projects/tree/master/nel-emerson/
- If you're interested in working with (the release candidate of) spaCy v3, an updated version of that code can be found here: https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson
- With spaCy v3, it becomes much easier to [swap out](https://nightly.spacy.io/usage/layers-architectures#swap-architectures) the [default](https://nightly.spacy.io/api/architectures#entitylinker) ML model for a custom one, and even to define a custom candidate generation algorithm `get_candidates`, see also [here](https://nightly.spacy.io/api/entitylinker#config).

**Comment by Joozty** [comment]
Thanks, Sofie, your presentation was really helpful and awesome at the same time! I will study more the links you provided. It's definitely more than I found through Google 😄
